app/models/google_calendar/event_editor.rb

```diff
@@ -0,0 +1,44 @@
+module GoogleCalendar
+  class EventEditor
+    def initialize(credentials)
+      @credentials = credentials
+      @calendar_service = GoogleCalendar::Client.new(credentials).calendar_service
+    end
+
+    def finish(conference_room_id, event_id)
+      @conference_room_email = ConferenceRoom.find(conference_room_id).email
+      @event = get_event(event_id)
+      finish_event
+    end
+
+    private
+
+    def get_event(event_id)
+      calendar_service.get_event(conference_room_email, event_id)
+    end
+
+    def update_event
+      calendar_service.update_event(conference_room_email, event.id, event)
+    end
+
+    def finish_event
+      return if event_not_started? || event_finished?
+      event.end.date_time = current_time
+      update_event
+    end
+
+    def event_not_started?
```

Comment:
Move this logic to `event` itself. Additionally, change these two methods to `event_in_progress?` and invert the condition. 

Please notice that now you check these two conditions against two different DataTime objects. It's not a problem here, but in general you shouldn't do this. It more robust to have these two conditions merged into one.